### DISCUSSION TITLE
How to add two numbers

### DISCUSSION THREAD

**Discussion by danielwilson1702** [community/community#25386] [Actions]
<p>Hello, how do I add a number to the GITHUB_RUN_NUMBER, this does not work:</p>
<pre><code>env:
      GITHUB_RUN_NUMBER_WITH_OFFSET: ${{ github.run_number + 200 }}

Unexpected symbol: '+'. Located at position 19 within expression: github.run_number + 200
</code></pre>
<p>Is it even possible to do a little computation like this? I’ve spent far too long trying to figure this out :slight_smile:</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p><a class="mention" href="/danielwilson1702">@danielwilson1702</a> </p>
<p>Github actions doesn’t support math operations in expressions inside ${{  }}. You could add up these two numbers in bash script and then use <strong>set-env</strong> command to give its value to an environment variable, please try my example: </p>
<pre><code>on: push

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - name: run number with offset
      env:
          NUM: ${{ github.run_number }}
      run: |
          echo ::set-env name=GITHUB_RUN_NUMBER_WITH_OFFSET::$(($NUM+200))
    - run: echo $GITHUB_RUN_NUMBER_WITH_OFFSET
</code></pre>

**Comment by danielwilson1702** [comment]
<p>Thank you, that works well!</p>

**Comment by renata-slalom** [comment]
<p>Hi, I tried a similar approach but it doesn’t seem to be working. Any ideas?</p>
<pre><code class="lang-auto">  - name: My Code
      env:
          OLDNUM: ${{ steps.my-step.outputs.mynum }}
      run: |
        echo "Current Number is $OLDNUM"
        echo ::set-env name=NEWNUM::$(($OLDNUM+1))
        echo "New Number is $NEWNUM"
</code></pre>
<p>It gets the output “mynum” correctly, I see it in the echo. But it doesn’t perform the operation of adding 1 to it or setting the new env var. I get an empty variable. Any ideas?</p>
<p>thanks</p>

**Comment by airtower-luna** [comment]
<p>An environment variable set using <code>::set-env</code> is not available within the same step, only the following ones (<a href="https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable" rel="nofollow noopener">see documentation</a>). If you want to use it in the same step you should be able to use normal Bash variable assignment.</p>

**Comment by Yanjingzhu** [comment]
<p>This is an expected behavior.</p>
<pre><code class="lang-auto">::set-env name={name}::{value}
</code></pre>
<p><em>This command creates or updates an environment variable for any actions running next in a job. The action that creates or updates the environment variable does not have access to the new value, but all subsequent actions in a job will have access.</em><br>
<a href="https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable" class="onebox" target="_blank" rel="nofollow noopener">https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable</a></p>
